html/semantics/embedded-content/media-elements/playing-the-media-resource/loop.html

```diff
@@ -0,0 +1,57 @@
+<!doctype html>
+<title>loop</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/media.js"></script>
+<div id="log"></div>
+<script>
+// Test the interaction of the seek algorithm's "Set the current playback
+// position to the given new playback position" with the loop step in "When the
+// current playback position reaches the end of the media resource when the
+// direction of playback is forwards". When the seek algorithm sets the current
+// playback position, that triggers the loop step which in turn triggers a new
+// seek. That seek aborts the first seek, which means that its timeupdate and
+// seeked events are never fired.
+function await_seeked(a, callback) {
```

Comment:
My first thought was that `EventWatcher` could be used for this test (and `promise_test` with actual `await`). Is there a reason you did it like this?

Reply:
Oh, this was written in 2014. That seems like a valid reason ^_^

Reply:
Yep, this test shows its age :)

Since this won't land verbatim I'll give it a shot to modernize when making the other changes.